src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs

```diff
@@ -290,6 +290,7 @@ private static CultureInfo GetCultureByName(string name)
         public static CultureInfo CreateSpecificCulture(string name)
         {
             CultureInfo? culture;
+            IcuGetPredefinedCultureInfo(name);
```

Comment:
This doesn't look correct to me at all. CreateSpecificCulture is intended to create a specific culture (to ensure it is not neutral culture). It doesn't matter if we have data for or not. it is same as any other culture we create.

Also You are calling Icu method while we could be running in NLS or Invariant mode.

Reply:
I want to add, we already have API `GetCultureInfo(string name, bool predefinedOnly)` to ensure creating predefined cultures only.